envi.go

```diff
@@ -183,6 +216,36 @@ func (envi *Envi) LoadYAMLFile(path string) error {
 	return nil
 }
 
+// LoadAndWatchYAMLFile loads key-value pairs from a yaml file,
+// then watches that file and reloads it when it changes.
+// Accepts an additional callback function that is executed
+// after the file was reloaded. Returns and error when something
+// goes wrong. When no error is returned, returns a close function
+// that should be deferred in the calling function.
+func (envi *Envi) LoadAndWatchYAMLFile(path string, callback func() error) (error, func() error) {
+	const errMessage = "failed to load and watch yaml file: %w"
+
+	err := envi.LoadYAMLFile(path)
+	if err != nil {
+		return fmt.Errorf(errMessage, err), nil
+	}
+
+	watcher, err := fsnotify.NewWatcher()
+	if err != nil {
+		return fmt.Errorf(errMessage, err), nil
+	}
+
+	go envi.fileWatcher(watcher, path, callback)
+
+	err = watcher.Add(path)
+	if err != nil {
+		watcher.Close()
+		return fmt.Errorf(errMessage, err), nil
+	}
+
+	return nil, watcher.Close
+}
```

Comment:
Ich glaub wir könnten das auch weiter abstrahieren, indem wir den function content in eine extra function geben und hier zusätzlich die loadfunction reingeben, dann sparen wir uns die code duplication